client/core/trade.go

```diff
@@ -842,38 +842,36 @@ func (t *trackedTrade) processCancelMatch(msgMatch *msgjson.Match) error {
 //
 // This method accesses match fields and MUST be called with the trackedTrade
 // mutex lock held for reads.
-func (t *trackedTrade) counterPartyConfirms(ctx context.Context, match *matchTracker) (have, needed uint32, changed, spent, expired bool) {
+func (t *trackedTrade) counterPartyConfirms(ctx context.Context, match *matchTracker) (have, needed uint32, changed, spent, expired bool, err error) {
+	fail := func(err error) (uint32, uint32, bool, bool, bool, error) {
+		return 0, 0, false, false, false, err
+	}
+
 	// Counter-party's swap is the "to" asset.
 	needed = t.wallets.toAsset.SwapConf
 
 	// Check the confirmations on the counter-party's swap. If counterSwap is
 	// not set, we shouldn't be here, but catch this just in case.
 	if match.counterSwap == nil {
-		t.dc.log.Warnf("counterPartyConfirms: No AuditInfo available to check!")
-		return
+		return fail(errors.New("counterPartyConfirms: No AuditInfo available to check!"))
 	}
 
 	wallet := t.wallets.toWallet
 	coin := match.counterSwap.Coin
 
 	_, lockTime, err := wallet.LocktimeExpired(match.MetaData.Proof.CounterContract)
 	if err != nil {
-		t.dc.log.Errorf("Error checking if locktime has expired on taker's contract on order %s, "+
-			"match %s: %v", t.ID(), match, err)
+		return fail(fmt.Errorf("error checking if locktime has expired on taker's contract on order %s, "+
+			"match %s: %w", t.ID(), match, err))
 	}
 	expired = time.Until(lockTime) < 0 // not necessarily refundable, but can be at any moment
 
 	have, spent, err = wallet.SwapConfirmations(ctx, coin.ID(),
 		match.MetaData.Proof.CounterContract, match.MetaData.Stamp)
 	if err != nil {
-		if !errors.Is(err, asset.ErrSwapNotInitiated) {
-			// No need to log an error if swap not initiated as this
-			// is expected for newly made swaps involving contracts.
-			t.dc.log.Errorf("Failed to get confirmations of the counter-party's swap %s (%s) "+
-				"for match %s, order %v: %v",
-				coin, t.wallets.toAsset.Symbol, match, t.UID(), err)
-		}
-		return
+		return fail(fmt.Errorf("failed to get confirmations of the counter-party's swap %s (%s) "+
+			"for match %s, order %v: %w",
+			coin, t.wallets.toAsset.Symbol, match, t.UID(), err))
```

Comment:
Makes sense, but the `LocktimeExpired` call above would have to error on `ErrSwapNotInitiated` because the zero-value for `lockTime` returned on error would be interpreted as "expired".  Which means it would never be `ErrSwapNotInitiated` down here because it would have been so above (for ETH at least).

An alternative might be to change the expired logic to `expired = !lockTime.IsZero() && time.Until(lockTime) < 0` so it never says expired=true when the lockTime is the zero value as is the case when `LocktimeExpired` has an error.  But I think I agree with Joe's conclusion that bubbling up an error is the least error prone way to keep the caller from jumping to conclusions about the actual state of the contract when it is actually unknown.

If you have a specific diff @buck54321 we could try talk over that?  I would like to get this fix in today in one form or another, and we could push a commit onto this PR if there's a revision that looks good.

Reply:
> Makes sense, but the `LocktimeExpired` call above would have to error on `ErrSwapNotInitiated`

I'm not suggesting changing anything up there. Just here to match previous behavior that wasn't broken.

Reply:
Sorry I must be misunderstanding your comment "only return an error if `!errors.Is(err, asset.ErrSwapNotInitiated)`".  Because when I mentally apply the suggestion, it would return for any error like it is in the present revision of this PR because it can't be `ErrSwapNotInitiated` because we're now catching it above (same contract call inside `wallet.LocktimeExpired`).
Apologies if I'm being dense.

Reply:
I see what you're saying now. The `ErrSwapNotInitiated` check down here is pointless then.

Reply:
Yea, eth should have already failed. I was going to do something like that at first, but after I was about to submit that pr I decided to do a little more that seems right to me and also fixes the bug. The returns should not be trusted if their is an error checking any of them, and the caller should be informed as of the reason of failure. The failure isn't that we are past locktime, the failure is that we couldn't check the locktime.

Reply:
imo was an easy bug to happen because of method design. Seeing the same thing in #1599 should just return errors and nil values on failure.